webpack/module-minifier-plugin/src/ModuleMinifierPlugin.ts

```diff
@@ -535,6 +545,29 @@ export class ModuleMinifierPlugin implements webpack.Plugin {
           }
         );
 
+        const { usePortableModules, compressAsyncImports } = this._optionsForHash;
+
+        function updateChunkHash(hash: Hash, chunk: webpack.compilation.Chunk): void {
+          // For now, automatically assume that any change to this plugin will alter the hash.
+          hash.update(pluginVersion as string);
+          // Using vs. not using source maps might affect output
+          hash.update(`${useSourceMaps}`);
```

Comment:
How difficult would it be to programmatically put all options into a stable structure and serializing and hashing that, so no one can forget to update this in the future?

Reply:
Agree with this. It's probably better to be over-cautious with what may affect the hash vs having to update this list every time a new setting that impacts the hash is added.

Reply:
I would need to update the minifier implementation to include a method `updateHash(hash)` that allows it to add its contribution. In the case of the `MessagePortMinifier` it will need to do a postmessage to the other end to get said hash contribution, which will need some testing. So most options can be handled via `JSON.stringify`, but there is also the issue that this plugin exposes its own hooks that affect code generation. That means a further `updateChunkHash` hook will need to be added to the API surface.

Doable, but mostly it kicks the can down to any extensions.

Reply:
`terser-webpack-plugin` uses a package called `serialize-javascript` to be able to serialize the TerserOptions structure, since it can contain arbitrary javascript. Likely, I would need to do the same.

Reply:
Updated to do the above.